### PR TITLE
3.1.0 beta

### PR DESCRIPTION
**Pull request by damianlegawiec**
- [X] more informative README 
- [X] capybara-screenshot added
- [X] dependencies updates, rubocop fixes
- [X] UX: always show add a review button

### CODE REVIEW

File: app/views/spree/shared/_reviews.html.erb [hunk 1]
```diff
@@ -8,7 +8,5 @@
       <%= render 'spree/shared/review', review: review %>
     <% end %>
   <% end %>
-  <% if !Spree::Reviews::Config[:require_login] || spree_current_user %>
```
Comment (priyank-gupta) [inline app/views/spree/shared/_reviews.html.erb]: @damianlegawiec Not sure why we have removed this check.

Comment (damianlegawiec) [inline app/views/spree/shared/_reviews.html.erb]: @priyank-gupta right now you have to sign in to add a review, visitors don't even know they can add a review, really bad UX. 

New flow: users clicks -> redirected to sign up (if Spree::Reviews::Config[:require_login] == tue) -> more registrations, more reviews, better UX, less confused visitors. Plus it's aligns better with full page caching.

File: spec/support/capybara.rb [hunk 2]
```diff
@@ -9,4 +10,6 @@
   Capybara.register_driver(:poltergeist) do |app|
     Capybara::Poltergeist::Driver.new app, js_errors: true, timeout: 60
   end
+
+  Capybara.save_and_open_page_path = ENV['CIRCLE_ARTIFACTS'] if ENV['CIRCLE_ARTIFACTS']
```
Comment (futhr) [inline spec/support/capybara.rb]: @damianlegawiec Hi, just curious how come you add Circle CI related stuff to the repos when they run on Travis and also lack circle.yml? By the way, nice work you guys do keeping the spree repositories maintained :+1:

Comment (damianlegawiec) [inline spec/support/capybara.rb]: @futhr nice catch, it shouldn't be in this repo or any other using Travis. Too little coffee, too many gems to fix ;)

File: app/views/spree/shared/_reviews.html.erb (hunk 1 above)
Comment (priyank-gupta) [inline app/views/spree/shared/_reviews.html.erb]: :+1:

**Comment by priyank-gupta** [comment]
:+1:

**Pull request merged**